packages/api/database/migrations/1605981580411_technology_order_chat.js

```diff
@@ -0,0 +1,20 @@
+/** @type {import('@adonisjs/lucid/src/Schema')} */
+const Schema = use('Schema');
+
+class TechnologyOrderChat extends Schema {
+	up() {
+		this.create('technology_order_chats', (table) => {
+			table.uuid('id').primary();
+			table.integer('technologyOrderId');
```

Comment:
It's important to use a foreign key here.
And I think it’s also interesting to continue using the snake case pattern as in the other schemas.
```suggestion
class TechnologyOrderChat extends Schema {
	up() {
		this.create('technology_order_chats', (table) => {
			table.uuid('id').primary();
			table
				.integer('technology_order_id')
				.unsigned()
				.references('id')
				.inTable('technology_orders')
				.onUpdate('cascade');
				.onDelete('cascade');
```

Reply:
fair enough, I will address your comments :)

Reply:
Além do padrão do plataforma em si, é uma conversão do Adonisjs. Na hora dos relacionamentos ele busca por padrão nometabela_id como chave estrangeira.

Reply:
@marcusmota essa implementação é específica para conversas na negociação da tecnologia? Não seria interessante ter algo mais escalável? Se precisar fazer um chat em outra situação a gente vai precisar criar outros esquemas.

Reply:
Sugestão interessante, @alexandreadames. Como seria essa implementação? Como diferenciaríamos quais são as mensagens de orders das demais?

Reply:
@mateus4k eu acho que a melhor forma de fazer isso seria criar uma entidade technology_chat, nessa entidade seriam armazenados os chats da tecnologia, e todas as mensagens referentes a tecnologia seriam respondidas pelo owner, sempre só teriam dois participantes owner e user_id:


technology_chat:
user_id: Perguntador,
technology_id: Id da tecnologia e apartir dai pegava o owner,
type: TECHNOLOGY_ORDER
status: active

Reply:
O type definiria a qual assunto se refere, nesse caso não precisaria de um order específico, nesse chat o cara iria perguntar sobre orders, digamos assim.

Reply:
E se fosse mais abrangente ainda? Um chat geral: seja pra tecnologia ou orders ou qualquer outra entidade 🤔

Reply:
TechnologyChat continua acoplado, não acha? sendo assim seria melhor criar um Chat e ChatMessages generico que teria um `object_id` e `object_type` ai para pegar os chats especificos das tecnologia order teria de passar o object_id e o object_type seria o tecnologyOrder.

Acho que fica bacana assim. o que acham?

Reply:
Por mim tudo bem, @marcusmota. Já resolve parcialmente o outro chat que teremos futuramente! 💯

Reply:
Muito boa ideia. Da certo assim.